### PR TITLE
[#5057] Add add_parent to hierarchy_parser (4-2-stable)

### PR DESCRIPTION
**Pull request by alanking**
Unit tests pass

### CODE REVIEW

**Comment by trel** [comment]
seems good - any other way to break it?   add parent can happen anywhere in the tree... so 'nonexistent' and 'bad input' seems to be the only things...    if good, please add #

File: lib/core/src/irods_hierarchy_parser.cpp [hunk 1]
```diff
@@ -87,10 +87,29 @@ namespace irods {
...
+        const std::string& _parent,
+        const std::string& _child) {
+        if (_parent == hierarchy_parser::delimiter()) {
+            THROW(SYS_INVALID_INPUT_PARAM, "invalid resource name");
+        }
+        auto it = resc_list_.begin();
+        if (!_child.empty()) {
+            it = std::find(resc_list_.begin(), resc_list_.end(), _child);
+            if (it == resc_list_.end()) {
+                THROW(CHILD_NOT_FOUND, (boost::format(
```
Comment (alanking) [inline lib/core/src/irods_hierarchy_parser.cpp:108]: Need to change this to use `fmt::format`... been sitting on this commit for quite some time!

**Comment by alanking** [comment]
I added a test for the beginning, end, middle, non-existent, and with invalid input (i.e. `;`) so I think that about covers it.

**Comment by trel** [comment]
good good.  # when ready

**Comment by alanking** [comment]
Done. Replaced `boost::format` usage with `fmt::format` as well

**Pull request merged**